ultrasonic/src/main/kotlin/org/moire/ultrasonic/fragment/SelectAlbumFragment.kt

```diff
@@ -117,6 +126,24 @@ class SelectAlbumFragment : Fragment() {
             false
         )
 
+        selectFolderHeader = SelectMusicFolderView(
+            requireContext(), albumListView!!,
+            MusicServiceFactory.getMusicService(requireContext()).getMusicFolders(
+                false, requireContext()
+            ),
+            activeServerProvider.getActiveServer().musicFolderId,
+            { _, selectedFolderId ->
+                if (!ActiveServerProvider.isOffline(context)) {
+                    val currentSetting = activeServerProvider.getActiveServer()
+                    currentSetting.musicFolderId = selectedFolderId
+                    serverSettingsModel.updateItem(currentSetting)
+                }
+                artistListModel.refresh(refreshAlbumListView!!)
```

Comment:
One note on this line. I really don't like that the SelectAlbumFragment now knows about the artist list. However the problem is that the CachedMusicServer offers no way to clear out the cache without changing the URL. So when a new folder is selected, the artists from the old folder are still in the cache. This is the only way I could find to force them to update. 

Suggestions on a better solution would be greatly appreciated.

Reply:
I think this can be solved in the checkSettingsChanged() in CachedMusicService. The musicFolderId can be stored similarly as the Url, and the cache can be deleted automatically if that changes.

Reply:
Thank You! This is an elegant solution, it works great.